### PR TITLE
Internal: Add REST to support pages panel UI for permissions [ED-13216]

### PR DESCRIPTION
**Pull request by ranshoshani34**

### CODE REVIEW

File: modules/site-navigation/rest-fields/page-user-can.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace Elementor\Modules\SiteNavigation\Rest_Fields;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+class Page_User_Can {
```
Comment (kobizz) [inline modules/site-navigation/rest-fields/page-user-can.php:9]: why do you need a different endpoint for user can?

Comment (ranshoshani34) [reply modules/site-navigation/rest-fields/page-user-can.php:9]: Its not an endpoint.
I'm adding a field to the 'pages' resource.
This field will be enriched when we request `/wp/v2/pages/?_fields=user_can`

**Review by kobizz — APPROVED** [review]

File: modules/site-navigation/data/endpoints/homepage.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+	public function get_items( $request ) {
+		$homepage_id = get_option( 'page_on_front' );
+		$show_on_front = get_option( 'show_on_front' );
+
+		if ( 'page' !== $show_on_front ) {
+			return 0;
+		}
+
+		return intval( $homepage_id );
```
Comment (blackcolt) [inline modules/site-navigation/data/endpoints/homepage.php:33]: return 'page' !== $show_on_front ? 0 : intval( $homepage_id )

**Review by blackcolt — DISMISSED** [review]

**Review by shai-shwartz — APPROVED** [review]

File: modules/site-navigation/data/endpoints/homepage.php (hunk 2 above)
Comment (StyleShit) [reply modules/site-navigation/data/endpoints/homepage.php:33]: Be positive instead 😉

```PHP
return 'page' === $show_on_front ? intval( $homepage_id ) : 0;
```

**Review by blackcolt — APPROVED** [review]

**Pull request merged**